drivers/dai/intel/dmic/dmic.c

```diff
@@ -876,14 +887,7 @@ const struct dai_driver_api dai_dmic_ops = {
 
 static int dai_dmic_initialize_device(const struct device *dev)
 {
-	IRQ_CONNECT(
-		DT_INST_IRQN(0),
-		IRQ_DEFAULT_PRIORITY,
-		dai_dmic_irq_handler,
-		DEVICE_DT_INST_GET(0),
-		0);
-
-	pm_device_init_suspended(dev);
```

Comment:
Co call to the new `dai_dmic_irq_init` function?

Reply:
Function is called inside of the dmic_pm_action callback. During the transition from OFF state to the SUSPEND (when power domain is powered-up).